### PR TITLE
Fix spring stiffness for BarReleases

### LINKED ISSUE #393: Stiffness of BarReleases is not correctly assigned to the ETABS model
#### Description:

When pushing bar release with spring stiffness, the stiffness values are not correctly assigned to the model.

Problem seem to stem from check being on if the dof type is Free instead of checking if it is fully Fixed.
#### Steps to reproduce:

Push bar with releases with spring stiffness, with DOF type set to spring.

#### Expected behaviour:

For releases with spring values and spring DOF type to be correctly pushed.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### NOTE: Depends on 

 ### Issues addressed by this PR

 Closes #393 

 

Makes sure the spring stiffness is picked up by changing the check if a DOF is released from checking if identical to free to instead check if it is not fixed.

 ### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/Evb6D2Mx5SlCldKS1XidhaUBl2Bi4gJ-lwtbV259ropqFg?e=0sPAc7

 ### Changelog

 ### Additional comments

### CODE REVIEW

**Review by JosefTaylor — CHANGES_REQUESTED** [review]
Needs similar change in ToBHoM in order to catch springs - see changes in SAP2000
Push works properly.

**Comment by IsakNaslundBh** [comment]
thanks @JosefTaylor .

Have fixed now!

File: Etabs_Adapter/CRUD/Read/Bar.cs [hunk 1]
```diff
@@ -148,41 +148,41 @@ private List<Bar> ReadBar(List<string> ids = null)
...
-            endRelease.TranslationalStiffnessX = endSpring[0];
-            endRelease.TranslationalStiffnessY = endSpring[1];
-            endRelease.TranslationalStiffnessZ = endSpring[2];
-            endRelease.RotationalStiffnessX = endSpring[3];
-            endRelease.RotationalStiffnessY = endSpring[4];
-            endRelease.RotationalStiffnessZ = endSpring[5];
-
-            BarRelease barRelease = new BarRelease() { StartRelease = startRelease, EndRelease = endRelease };
+            Constraint6DOF bhStartRelease = new Constraint6DOF();
+
```
Comment (JosefTaylor) [inline Etabs_Adapter/CRUD/Read/Bar.cs:154]: Push uses XZY, pull uses XYZ- I am not sure which one is right! I think it is XZY.

**Review by JosefTaylor — CHANGES_REQUESTED** [review]
axes are switched between pulling and pushing

**Comment by IsakNaslundBh** [comment]
> axes are switched between pulling and pushing: Push uses XZY, pull uses XYZ- I am not sure which one is right! I think it is XZY.

You are right 🤦‍♂️ . Have fixed now, thanks for spotting.

Also added another section of the script to verify that the order is correct by looking at reactions, and as you stated, XZY looks to be correct. Really confusing with Etabs Left-hand rule with switched orientations between 1 and 2 compared to BHoM (and what I have seen in most other packages).

Anyhow, code updated and added a small method over the quite long statement of ?s and :s as I found that a bit hard to read if we look at this again :)

**Review by JosefTaylor — APPROVED** [review]
Works great!

**Comment by JosefTaylor** [comment]
I have this transform matrix in SAP2000; is there a way we could use it to create releases?
`        public static TransformMatrix barLocalAxisToBHoM = new TransformMatrix()
        {
            Matrix = new double[4, 4]
            {
                { 1,  0,  0, 0 },
                { 0,  0, -1, 0 },
                { 0,  1,  0, 0 },
                { 0,  0,  0, 1 }
            }
        };`

**Comment by IsakNaslundBh** [comment]
@BHoMBot check required

**Comment by IsakNaslundBh** [comment]
@BHoMBot check copyright-compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check required

**Pull request merged**